.github/workflows/ci-macvim.yaml

```diff
@@ -258,8 +279,19 @@ jobs:
           # CI environment.
           make -C src macvim-dmg CREATEDMG_FLAGS=--skip-jenkins
 
+          if ${{ matrix.legacy == true }}; then
+            mv src/MacVim/build/Release/MacVim.dmg src/MacVim/build/Release/MacVim_10.9.dmg
+          fi
+
+      - name: Upload MacVim (legacy) image
+        if: startsWith(github.ref, 'refs/tags/') && matrix.publish && matrix.legacy
+        uses: actions/upload-artifact@v3
+        with:
+          name: MacVim_10.9.dmg
+          path: src/MacVim/build/Release/MacVim_10.9.dmg
```

Comment:
Does it need to rename to `MacVim_10.9.dmg` at test step?
I think can set `name` to `MacVim_10.9.dmg` and `path` to `src/MacVim/build/Release/MacVim.dmg` at upload step.
Additionally, can aggregate upload steps as setting name to `MacVim${{ matrix.legacy ? "_10.9" : "" }}.dmg`
(but I don't make sure it works); does it feel complicated?

Reply:
> Does it need to rename to `MacVim_10.9.dmg` at test step?
> I think can set `name` to `MacVim_10.9.dmg` and `path` to `src/MacVim/build/Release/MacVim.dmg` at upload step.

The `name` part of `upload-artifact` only defines what the artifact looks like in GitHub Actions. Once you download the file, it uses the original file name, so if we didn't do the rename, it will still shows up as `MacVim.dmg`. I couldn't find a way to remap the file name in the available parameters. I personally have to sign / notarize the app off CI anyway (since I never set up secrets sharing with GitHub Actions, which I'm not sure if I want to do), so I can still rename them when I do that, but it feels like the output of the CI build step should result in a file called `MacVim_10.9.dmg`.

Reply:
> Additionally, can aggregate upload steps as setting name to `MacVim${{ matrix.legacy ? "_10.9" : "" }}.dmg`
> (but I don't make sure it works); does it feel complicated?

Huh. I can give it a try. I think I couldn't get the syntax to quite work out but I don't think I tried the conditional that you showed. I do dislike how we have to have two separate step for uploading artifacts. I'm not sure if it will work though.

Reply:
> The name part of upload-artifact only defines how the artifact is named in GitHub Actions. Once you download the file (either through the web, or the gh command-line tool), it uses the original file name. So basically if we didn't do the rename, the downloaded file will still show up as MacVim.dmg. I couldn't find a way to remap the file name in the available parameters.

I see, thank you.

> > Additionally, can aggregate upload steps as setting name to MacVim${{ matrix.legacy ? "_10.9" : "" }}.dmg (but I don't make sure it works); does it feel complicated?
> 
> Huh. I can give it a try. I think I couldn't get the syntax to quite work out but I don't think I tried the conditional that you showed. I do dislike how we have to have two separate step for uploading artifacts. I'm not sure if it will work though.
> 

Sorry, GitHub Actions does not compatible with ternary operator, but I confirmed can use `MacVim${{ matrix.legacy && '_10.9' || '' }}.dmg`.

Reply:
I wonder if we should just make a new matrix variable called `matrix.file_postfix`. This way we just define it in one place instead of having to paste "_10.9" all over the place, and less reliance on complicated logic.

Reply:
Alright just pushed an update to use a matrix variable instead, so it's easier to modify later. Here's an example run (testing was removed to make it run faster): https://github.com/ychin/macvim/actions/runs/3366660777

Reply:
Let me know if you have more comments.

Reply:
I feel the variable `matrix.legacy` is less necessary due to `matrix.publish_postfix`.

Reply:
Hmm, I see what you mean. Functionally yes, but I think I prefer keeping them separate for now, as they have different meanings. For example, let's say I actually want to publish MacVim as `MacVim_10.13.app` instead I can just add `publish_postfix` to the matrix and it would work, without touching the legacy parts.